Source/ACE/Entity/Player.cs

```diff
@@ -2516,6 +2521,49 @@ public ActionChain CreateMoveToChain(ObjectGuid target, float distance)
             return moveToChain;
         }
 
+        public ActionChain CreateCollideWithChain(ObjectGuid target)
```

Comment:
If I understand correctly, this large chunk of copy/pasted code is largely a hack to work around portal distances being -0.1.

This seems like a very convoluted way to solve that problem. Why can't you do something simpler, like change the distance calculations in MoveTo to check against a radius instead of radius squared? Or switch portals onuse distance to 0?

I still don't understand why you call HandleActionOnCollide -- that seems wrong.